src/h5web/explorer/Explorer.tsx

```diff
@@ -18,25 +18,53 @@ function Explorer(props: Props): JSX.Element {
 
   const [selectedNode, setSelectedNode] = useState<TreeNode<HDF5Link>>();
 
+  useEffect(() => {
+    if (tree) {
+      // Select root node when tree is ready
+      setSelectedNode(tree);
+    }
+  }, [tree]);
+
   useEffect(() => {
     if (selectedNode) {
+      // Propagate selected link to parent component
       onSelect(selectedNode.data);
     }
   }, [selectedNode, onSelect]);
 
+  if (!tree) {
+    return (
+      <div className={styles.explorer}>
+        <p className={styles.domain}>{domain}</p>
+        <p className={styles.loading}>Loading...</p>
+      </div>
+    );
+  }
+
   return (
-    <div className={styles.explorer}>
-      <p className={styles.domain}>{domain}</p>
-
-      <TreeView
-        nodes={tree}
-        selectedNode={selectedNode}
-        isRoot
-        onSelect={setSelectedNode}
-        renderIcon={(data, isBranch, isExpanded) => (
-          <Icon data={data} isBranch={isBranch} isExpanded={isExpanded} />
-        )}
-      />
+    <div className={styles.explorer} role="tree">
+      <button
+        className={styles.domainBtn}
+        type="button"
+        role="treeitem"
+        aria-selected={selectedNode === tree}
+        onClick={() => {
+          setSelectedNode(tree);
+        }}
+      >
+        {domain}
+      </button>
```

Comment:
Even though the domain is part of the tree data structure, I decided not to render it within `TreeView` to avoid having to  deal with a root case in the recursion (`div` instead of `ul`, no expansion, no icon, etc.)